llvm/lib/SYCLLowerIR/SanitizeDeviceGlobal.cpp

```diff
@@ -43,12 +43,22 @@ static bool instrumentDeviceGlobal(Module &M) {
   //  size_t beginning address of the device global
   StructType *StructTy = StructType::get(IntTy, IntTy, IntTy);
 
+  const GlobalVariable *OMPOffloadEntry =
+      M.getNamedGlobal(".omp_offloading.entry_name");
```

Comment:
It's probably more reliable to use `M.getModuleFlag("openmp-device");` to check for OpenMP device compilation. See this for reference: https://github.com/intel/llvm/blob/2adf19646e48060f9d1a8fd143ed1276b4967c77/llvm/lib/Transforms/IPO/OpenMPOpt.cpp#L5944

Reply:
Hi, @abhinavgaba 
Done.
Thanks very much!